echopype/calibrate/calibrate_ek.py

```diff
@@ -102,14 +106,13 @@ def get_cal_params(self, cal_params, waveform_mode, encode_mode):
         cal_params : dict
         """
 
-        if (
-            encode_mode == "power"
-            and waveform_mode == "CW"
-            and self.echodata.beam_power is not None
-        ):
-            beam = self.echodata.beam_power
+        try:
+            if encode_mode == "power" and waveform_mode == "CW":
+                beam = self.echodata["Sonar/Beam_group2"]
+        except GroupNotFoundError:
+            beam = self.echodata["Sonar/Beam_group1"]
         else:
-            beam = self.echodata.beam
+            beam = self.echodata["Sonar/Beam_group1"]
```

Comment:
It seems that this should be the following?
```
if encode_mode == "power" and waveform_mode == "CW":
   try:
      beam = self.echodata["Sonar/Beam_group2"]
   except GroupNotFoundError:
      beam = self.echodata["Sonar/Beam_group1"]
else 
   beam = self.echodata["Sonar/Beam_group1"]
```

The logic I am following is that
- `Sonar/Beam_group2` only exists when there are both power and complex data, and in this case:
   - power data will be in `Sonar/Beam_group2`
   - complex data will be in `Sonar/Beam_group1`
- If only power or only complex data exist, they will be in `Sonar/Beam_group1`

Did I misunderstand this?

Reply:
Resolved. Per discussion, `__getitem__` has change to return `None` so `try/except` is not needed and changes is simply on the access pattern.